components/badge/badge.vue

```diff
@@ -2,19 +2,42 @@
   <span
     :class="[
       'd-badge',
-      BADGE_COLOR_MODIFIERS[color],
+      BADGE_TYPE_MODIFIERS[type],
+      BADGE_KIND_MODIFIERS[kind],
     ]"
     data-qa="dt-badge"
   >
-    <!-- @slot Slot for badge content, defaults to text prop -->
-    <slot>
-      {{ text }}
-    </slot>
+    <span
+      v-if="$slots.iconLeft || type === 'ai'"
+      class="d-badge__icon-left"
+    >
+      <!-- @slot Slot for an icon to the left of the badge, insert a dt-icon -->
+      <slot name="iconLeft">
+        <dt-icon
+          name="dialpad-ai"
+          size="200"
+        />
+      </slot>
+    </span>
+    <span class="d-badge__label">
+      <!-- @slot Slot for badge content, defaults to text prop -->
+      <slot>
+        {{ text }}
+      </slot>
+    </span>
+    <span
+      v-if="$slots.iconRight"
+      class="d-badge__icon-right"
```

Comment:
@modialpad Both `kind`s, `label` and `count`, must support any of our icons. Dialtone docs list [some examples](https://dialpad.design/deploy-previews/pr-764/components/badge.html#examples).

Reply:
@modialpad You're saying just have a leftIcon/rightIcon prop where you pass in the icon name rather than a slot? This is probably a good idea since this component is very fixed size. This would of course prevent any customization of the icon.. which might be what we want.

Reply:
Ah, that makes sense when you put it that way.

Reply:
Made the change to use prop instead of slot

Reply:
> @modialpad You're saying just have a leftIcon/rightIcon prop where you pass in the icon name rather than a slot? This is probably a good idea since this component is very fixed size. This would of course prevent any customization of the icon.. which might be what we want.

Funny enough I actually meant it the way Francis interpreted, basically hardcoding the icons inside if we know all the use-cases, but I'm glad it ended up for the best haha. I agree, using `DtIcon` is a better approach for this.